doc/tutorials/introduction/test_writing/accuracy_test_writing.markdown

```diff
@@ -0,0 +1,269 @@
+
+## ACCURACY TESTS
+
+### Work directory
+
+All modules have their own dir for accuracy tests: `opencv_contrib/modules/(moduleName)/test/...`
+
+### Dir Structure
+
+- `test_precomp.hpp` - file for includes
+- `test_main.cpp` - main file of test sample
+- `test_smth1.cpp` - files for tests
+- `test_smth2.cpp`
+- `...`
+
+### Test structure
+
+```c++
+// name of this case is "name1.name2"
+TEST(name1, name2)
+{
+    ASSERT_....;
+}
+```
+
+### Tests sample example
+
+**Files:**
+
+- test_precomp.hpp
+- test_main.cpp
+- test_sum.cpp
+- test_sub.cpp
+
+
+
+```c++
+// test_precomp.hpp
+
+#ifndef __OPENCV_TEST_PRECOMP_HPP__
+#define __OPENCV_TEST_PRECOMP_HPP__
+
+#include <...>
+#include "..."
+
+namespace opencv_test {
+  using namespace cv::(moduleName);
+}
+
+#endif
+```
+
+
+
+```c++
+// test_main.cpp
+#include "test_precomp.hpp"
+
+CV_TEST_MAIN("<FOLDER_NAME_IN_TESTDATA>")
+```
+
+
+
+```c++
+// test_sum.cpp
+
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+TEST(SUM, int)
+{
+    int a=1;
+    int b=2;
+    int res=a+b;
+    ASSERT_EQ(res, 3)
+}
+
+TEST(SUM, float)
+{
+    float a=0.1f;
+    float b=0.2f;
+    float res=a+b;
+    ASSERT_EQ(res, 0.3f)
+}
+
+}} // namespace
+```
+
+
+
+
+```c++
+// test_sub.cpp
+
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+TEST(SUB, int)
+{
+    int a=2;
+    int b=1;
+    int res=a-b;
+    ASSERT_EQ(res, 1)
+}
+
+TEST(SUB, float)
+{
+    float a=0.2f;
+    float b=0.1f;
+    float res=a-b;
+    ASSERT_EQ(res, 0.1f)
+}
+
+}} // namespace
+```
+
+
+### Run
+
+Before the first run of new tests, you need to rebuild the project with CMake.
+
+If you want to run **ALL TESTS**, you need to run the test sample or use your system:
+
+- (*Windows*) move to the folder `<BUILD_DIR>/bin/Debug(Release)/` and run the executive file `opencv_test_<MODULE_NAME>.exe`
+- (*Linux*)
+
+If you want to run **CERTAIN TESTS** (SUM.int test, for example).
+
+- If you work in IDE you need to find the project preference named `Command Arguments` or similar and write there the key like `--gtest_filter=<KEY>` (KEY = SUM.int in our example).  [gtest_filter.doc](--gtest_filter)
+- If you want to use your system:
+  - (*Windows*) move to the folder `<BUILD_DIR>/bin/Debug(Release)/` and run the executive file in the console `opencv_test_<MODULE_NAME>.exe --gtest_filter=<KEY>` (KEY = SUM.int in our example).
+  - (*Linux*)
+
+
+
+# TEST DATA
+
+Repository with the test data (opencv_extra): https://github.com/opencv/opencv_extra
+
+Before working with data you need to set environment variable `OPENCV_TEST_DATA_PATH`:
+
+- (Windows)
+  1. Win+R
+  2. Sysdm.cpl
+  3. Go to Advanced -> Environment Variables -> System variables (New..)
+  4. Set name as `OPENCV_TEST_DATA_PATH` and value as `<PATH_TO_opencv_extra>/testdata`
+- (Visual Studio)
+  1.  Go to Project -> Properties -> Debugging -> Environment
+  2. Set the Environment as `OPENCV_TEST_DATA_PATH=<PATH_TO_opencv_extra>/testdata`
+- (Linux)
```

Comment:
The first sentence is enough. With addition `=<opencv_extra>/testdata`

Other information are questionable and inaccurate.
We should write documentation about OpenCV specific instead of trying to cover general programming topics.

Reply:
done